testing/test_faulthandler.py

```diff
@@ -98,6 +98,33 @@ def test_cancel_timeout_on_hook(monkeypatch, hook_name):
 
     # call our hook explicitly, we can trust that pytest will call the hook
     # for us at the appropriate moment
-    hook_func = getattr(plugin_module, hook_name)
-    hook_func()
+    hook_func = getattr(FaultHandlerHooks, hook_name)
+    hook_func(self=None)
     assert called == [1]
+
+
+def test_already_initialized(testdir):
+    """Test for faulthandler being initialized earlier than pytest (#6575)"""
+    testdir.makepyfile(
+        """
+        def test():
+            import faulthandler
+            assert faulthandler.is_enabled()
+    """
+    )
+    result = testdir.run(
+        sys.executable,
+        "-X",
+        "faulthandler",
+        "-mpytest",
+        testdir.tmpdir,
+        "-o",
+        "faulthandler_timeout=20",
+    )
+    result.stdout.fnmatch_lines(
+        [
+            "* PytestConfigWarning: faulthandler module enabled before pytest configuration*",
+            "*1 passed*",
```

Comment:
Note that the test fails with `-o faulthandler_timeout=0` - not sure if that is good/expected.

Reply:
Thanks for double checking, but it is expected: the test only "fails" because we don't produce any warning, but there's no error. 

I will update the test to cover that case though.

Reply:
Hardened the test. 👍

Reply:
Ok.  I wasn't sure however if specifying `timeout=0` should still enable it though.  But it seems like this is also a way to disable it.